src/guide/agent-functions/defining-agent-functions.rst

```diff
@@ -74,12 +84,128 @@ When using the Run-Time Compiled (RTC) functions, optionally in the C++ API or r
+
+    #Define an agent function called agent_fn1
+    @pyflamegpu.agent_function
+    def outputdata(message_in: pyflamegpu.MessageNone, message_out: pyflamegpu.MessageNone):
+        # Behaviour goes here
+        pass
+
+    ...
+    # Transpile the Pure Python agent function to equivalent C++ (and transpile errors with raise an exception at this stage)
+    agent_fn1_translated = pyflamegpu.codegen.translate(agent_fn1) 
+
+    # Attach pure Python function called agent_fn1 to an agent represented by the AgentDescription agent (the function will be compiled at runtime as C++)
+    agent.newRTCFunction("agent_fn1", agent_fn1_translated)
+    ...
+    
+Supported Function Calls
+""""""""""""""""""""""""
+
+The pure Python transpiler supports function calls that have a C++ equivalent in the will check function calls in the API. The API singleton is available as ``pyflamegpu`` which is equivalent to the ``FLAMEGPU`` object in C++. If non existent functions are called on API objects then this will raise an error during translation. Calling of correctly specified device functions is also supported as well as calls to a small number of python built in functions. E.g. ``abs()``, ``int()``, and ``float()``. Any calls to math library functions are directly translated to C++ equivalents. E.g. ``Math.sinf()`` will result in a call to ``sinf()`` in C++. The majority of python math functions directly translate to their C++ counterparts. 
+
+Supported Math Constants
+""""""""""""""""""""""""
+
+The following mathematics constants are supported;
+
+============= =================================
+Python        Description
+============= =================================
+``math.pi``   Translated of ``M_PI`` in C++
+``math.e``    Translated of ``M_E`` in C++
+``math.inf``  Translated of ``INFINITY`` in C++
+``math.nan``  Translated of ``NAN`` in C++
+============= =================================
+
+.. _Pure Python Types:
+
+Typed API Functions
+"""""""""""""""""""
+
+Calls to many FLAME GPU API functions are typed using template arguments in C++. In the Pure python versions equivalents the type is specified by calling a type instantiated version of the function using a suffix. The following type suffix are supported;
+
+============= ===============================================
+Type Suffix   C++ Equivalent type
+============= ===============================================
+``Char``      Would be ``char`` type template argument 
+``Float``     Would be ``float`` type template argument 
+``Double``    Would be ``double`` type template argument 
+``Int``       Would be ``int`` type template argument 
+``UInt``      Would be ``unsigned int`` type template argument 
+``Int8``      Would be ``int_8`` type template argument 
+``UInt8``     Would be ``uint_8`` type template argument 
+``Int16``     Would be ``int_16`` type template argument 
+``UInt16``    Would be ``uint_16`` type template argument 
+``Int32``     Would be ``int_32`` type template argument 
```

Comment:
Might want to point out that `Int32` is the same as `Int` (and same for unsigned)

Reply:
Done